### PR TITLE
add windows latest and gcc 8 builds to azure

### PR DESCRIPTION
**Pull request by phlptp**
Add a MSVC 2019 build and a gcc 8 build for use with C++17.

Test issue #445

### CODE REVIEW

**Comment by phlptp** [comment]
Why is azure not running?

**Comment by henryiii** [comment]
Good question. Last run is on Wednesday.

**Comment by henryiii** [comment]
I think it might just be a hiccup? You could try repushing. I'll add you to the Azure admins if you PM me your azure ID (username or whatever it uses).

**Comment by phlptp** [comment]
So as far as I can tell something must have changed in azure pipelines that they don't automatically run on PR's unless you specify that in the config file.  

I added three tests.  a windows latest for visual studio 2019, which It didn't look like we were testing otherwise and probably should.  
A gcc 8 C++17 which is the trigger for this that was failing and is now passing.  

and a clang8 C++17 build which I thought might fail for the same reason as GCC 8 but didn't,  So not sure what is up with that yet.  

We can leave in the new clang8 build or not.   The fix was to specifically target GCC 8 and C++17 and add the specific header.  

I probably should add a build note to the read me about the potential issue yet.

**Comment by phlptp** [comment]
fixes #445

File: CMakeLists.txt [hunk 1]
```diff
@@ -143,6 +143,11 @@ target_compile_options(CLI11_warnings
 add_library(CLI11 INTERFACE)
 add_library(CLI11::CLI11 ALIAS CLI11) # for add_subdirectory calls
 
+if (CMAKE_CXX_STANDARD GREATER 16)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 8)
```
Comment (henryiii) [inline CMakeLists.txt]: Why is this "not" version equal 8? This only needs to be added for C++17 & GCC < 9.

**Comment by henryiii** [comment]
Wouldn't it be better to modify the automatic selection of the filesystem header to not turn on for GCC < 9? Then a user could force it on, but if not, then it doesn't try to use it. That also fixes the single file header.

File: CMakeLists.txt (hunk 1 above)
Comment (phlptp) [reply CMakeLists.txt]: I think it only really applies to GCC 8.  GCC 7 doesn't have <filesystem> and GCC 9 doesn't have a problem, so this was a the condition that needed the extra linking target.

**Comment by phlptp** [comment]
We would have to figure out the condition for the libstdc++ that matched Gcc 8, but it could be done that way as well.   I suppose from a single header perspective that would be better since it wouldn't need the extra linking library.

File: CMakeLists.txt (hunk 1 above)
Comment (henryiii) [reply CMakeLists.txt]: But shouldn't it be `if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 8)`?

Comment (phlptp) [reply CMakeLists.txt]: Yes, so why is then working on gcc 8?   I am trying with the in code check, if that works we can probably remove it from CMAKE.

Comment (henryiii) [reply CMakeLists.txt]: The version equal bothers me; `VERSION_LESS 9` would be better. I assume this isn't expecting 8.0.0?

**Comment by henryiii** [comment]
Looks good; does the special instructions need to be updated?

Also, maybe of interest to @cbachhuber, I think we should add `IndentPPDirectives: AfterHash` to make this easier to read. (not in this PR, though).

**Comment by phlptp** [comment]
I did update the readme note regarding gcc 8,  if you think it needs to be clearer we can edit it.

**Pull request merged**